posthog/models/event.py

```diff
@@ -98,6 +98,30 @@ def __init__(self, selector: str, escape_slashes=True):
             part.unique_order = len([p for p in self.parts if p.data == part.data])
             self.parts.append(copy.deepcopy(part))
 
+    def _split(self, selector):
+        in_attribute_selector = False
+        in_quotes: Optional[str] = None
+        part: List[str] = []
+        for char in selector:
+            if char == "[" and in_quotes is None:
+                in_attribute_selector = True
+            if char == "]" and in_quotes is None:
+                in_attribute_selector = False
+            if char in "\"'":
+                if in_quotes is not None:
```

Comment:
Small readability nit:

```suggestion
            if char == "[" and not in_quotes:
                in_attribute_selector = True
            if char == "]" and not in_quotes:
                in_attribute_selector = False
            if char in "\"'":
                if not in_quotes:
```

Reply:
I'd rather keep it this way. Whether a value is truthy or falsy varies from language-to-language and is a frequent pitfall when acquainting with unfamiliar codebases.

See also google style guide: https://google.github.io/styleguide/pyguide.html#2144-decision